codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttSubAckPayload.java

```diff
@@ -28,39 +28,43 @@
  */
 public class MqttSubAckPayload {
 
-    private final List<Integer> grantedQoSLevels;
+    private final List<Integer> reasonCodes;
 
-    public MqttSubAckPayload(int... grantedQoSLevels) {
-        ObjectUtil.checkNotNull(grantedQoSLevels, "grantedQoSLevels");
+    public MqttSubAckPayload(int... reasonCodes) {
+        ObjectUtil.checkNotNull(reasonCodes, "reasonCodes");
 
-        List<Integer> list = new ArrayList<Integer>(grantedQoSLevels.length);
-        for (int v: grantedQoSLevels) {
+        List<Integer> list = new ArrayList<Integer>(reasonCodes.length);
+        for (int v: reasonCodes) {
             list.add(v);
         }
-        this.grantedQoSLevels = Collections.unmodifiableList(list);
+        this.reasonCodes = Collections.unmodifiableList(list);
     }
 
     public MqttSubAckPayload(Iterable<Integer> grantedQoSLevels) {
-        ObjectUtil.checkNotNull(grantedQoSLevels, "grantedQoSLevels");
+        ObjectUtil.checkNotNull(grantedQoSLevels, "reasonCodes");
         List<Integer> list = new ArrayList<Integer>();
         for (Integer v: grantedQoSLevels) {
             if (v == null) {
                 break;
             }
             list.add(v);
         }
-        this.grantedQoSLevels = Collections.unmodifiableList(list);
+        this.reasonCodes = Collections.unmodifiableList(list);
     }
 
     public List<Integer> grantedQoSLevels() {
-        return grantedQoSLevels;
+        return reasonCodes;
```

Comment:
@paul-lysak hmm... this feels not correct to me. Do we need to use `& 0x03` for the values here ?

Reply:
In fact, it already can return error code `0x80` beside the actual QoS levels. Nevertheless, I've added a mask here to ensure no new error codes.

Reply:
can you add a test case for this as well ?

Reply:
done